sound/soc/sof/intel/hda.c

```diff
@@ -1039,13 +1039,59 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 #endif
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_SOUNDWIRE)
+/* Check if all slaves defined on the link can be found */
+static bool is_slave_found(struct snd_sof_dev *sdev,
+			   struct snd_soc_acpi_link *link,
+			   struct sdw_intel_ctx *sdw)
```

Comment:
the name of the function does not represent what it does: it checks if all expected slaves are present on a link, not just a single check.

Maybe link_slaves_found() ?